src/modules/commander/Commander.cpp

```diff
@@ -2244,7 +2392,9 @@ Commander::run()
 				tune_negative(true);
 			}
 
-			if (_manual_control_switches_sub.update(&_manual_control_switches) || safety_updated) {
+			_controller_is_a_joystick = _manual_control_setpoint.data_source > manual_control_setpoint_s::SOURCE_RC;
```

Comment:
You shouldn't need to store any state (_controller_is_a_joystick) as the manual_control_setpoint is always available where this is checked.

Reply:
good point